h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/task/XGBoostSaveMatrixTask.java

```diff
@@ -0,0 +1,50 @@
+package hex.tree.xgboost.task;
+
+import ml.dmlc.xgboost4j.java.DMatrix;
+import ml.dmlc.xgboost4j.java.XGBoostError;
+import hex.tree.xgboost.matrix.MatrixLoader;
+import org.apache.log4j.Logger;
+import water.H2O;
+import water.Key;
+
+import java.io.File;
+import java.io.IOException;
+
+public class XGBoostSaveMatrixTask extends AbstractXGBoostTask<XGBoostSaveMatrixTask> {
+
+    private static final Logger LOG = Logger.getLogger(XGBoostSaveMatrixTask.class);
+
+    private final MatrixLoader _matrixLoader;
+    private final String _saveMatrixDirectory;
+    
+    protected transient DMatrix matrix;
+    
+    public XGBoostSaveMatrixTask(Key modelKey, String saveMatrixDirectory, boolean[] hasDMatrix, MatrixLoader loader) {
+        super(modelKey, hasDMatrix);
+        _saveMatrixDirectory = saveMatrixDirectory;
+        _matrixLoader = loader;
+    }
+
+    @Override
+    protected void execute() {
+        try {
+            matrix = _matrixLoader.makeLocalMatrix();
```

Comment:
Do I understand it correctly that the main cluster still creates a DMatrix?

Reply:
yes, this might be one thing I may want to improve in the future, rather than saving the dmatrix, save the intermediate java in-memory data

Reply:
Okay, this is fine for the development version but IMHO it is a must-have before we can give this to users. It would mean we couldn't really avoid the xgboost overhead parameter on the main cluster.

Reply:
done (with some TODO)